pkg/kubelet/dockershim/docker_image.go

```diff
@@ -98,10 +98,15 @@ func (ds *dockerService) RemoveImage(image *runtimeapi.ImageSpec) error {
 			}
 		}
 		return nil
+	} else if err != nil && dockertools.IsImageNotFoundError(err) {
+		return nil
 	}
 
 	_, err = ds.client.RemoveImage(image.Image, dockertypes.ImageRemoveOptions{PruneChildren: true})
-	return err
+	if err != nil && !isImageNotFoundError(err) {
```

Comment:
The dockertools package is going away, but most of the utility code will be moved to dockershim instead.
I have a PR to migrate those code: https://github.com/kubernetes/kubernetes/pull/45307